### PR TITLE
[IMP] l10n_ec: new income withholding taxes for EC

### PR DESCRIPTION
**Pull request by rrvc12**
In 2024 the SRI changes income withholding taxes. We update minor data and create new taxes for the new percentages

### CODE REVIEW

File: addons/l10n_ec/data/account.account.template.csv [hunk 1]
```diff
@@ -151,12 +151,13 @@ ret_ir_1x100,2114010201,Retenciones de la fuente 1%,FALSE,liability_current,l10n
...
-ret_ir_15x100,2114010208,Retenciones de la fuente 15%,FALSE,liability_current,l10n_ec_ifrs
-ret_ir_22x100,2114010209,Retenciones de la fuente 22%,FALSE,liability_current,l10n_ec_ifrs
-ret_ir_others,2114010210,Otras retenciones,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_3x100,2114010205,Retenciones de la fuente 3%,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_5x100,2114010206,Retenciones de la fuente 5%,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_8x100,2114010207,Retenciones de la fuente 8%,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_10x100,2114010208,Retenciones de la fuente 10%,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_15x100,2114010209,Retenciones de la fuente 15%,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_22x100,2114010210,Retenciones de la fuente 22%,FALSE,liability_current,l10n_ec_ifrs
+ret_ir_others,2114010211,Otras retenciones,FALSE,liability_current,l10n_ec_ifrs
```
Comment (pepetreshere) [inline addons/l10n_ec/data/account.account.template.csv:160]: Move the new account to the end

File: addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.account.models.chart_template import update_taxes_from_templates
+from odoo import api, SUPERUSER_ID
+
+def update_names(cr, env):
```
Comment (jco-odoo) [inline addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py:6]: Should not this happen by the update_taxes already?

File: addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            (f'{company.id}_tax_withhold_profit_308', '308 10% Utilizacion o Aprovechamiento de la Imagen o Renombre (Personas Naturales,Sociedades,"Influencers")'),
+            (f'{company.id}_tax_withhold_profit_3440', '3440 2.75% Otras Retenciones Aplicables el 2,75%')
+        ]
+        for xml_id, name in taxes_to_update:
+            cr.execute("UPDATE account_tax SET name=%s "
+                       "WHERE id=(SELECT res_id FROM ir_model_data WHERE module = 'l10n_ec' AND name=%s)", (name, xml_id))
+
+def update_ec_codes(cr, env):
+    """
+    Update special fields for Ecuador, the code base and code applied in the taxes
```
Comment (jco-odoo) [inline addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py:23]: This should happen automatically as well?

File: addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        for xml_id in taxes_to_inactivate:
+            cr.execute("UPDATE account_tax SET active=False "
+                       "WHERE id=(SELECT res_id FROM ir_model_data WHERE module = 'l10n_ec' AND name=%s)", (xml_id,))
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+
+    update_taxes_from_templates(cr, 'l10n_ec.l10n_ec_ifrs')
+    update_names(cr, env)
+    update_ec_codes(cr, env)
+    inactivate_taxes_replaced(cr, env)
```
Comment (jco-odoo) [inline addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py:64]:
```suggestion
    update_taxes_from_templates(cr, 'l10n_ec.l10n_ec_ifrs')
```
What does not work if you just do this?

Comment (rrvc12) [reply addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py:64]: In 'update_taxes_from_templates' method when it detects a tax that already exists, it only updates the tax tags and nothing more...So I update the rest in other methods.

Comment (jco-odoo) [reply addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py:64]: @rrvc12 Ok, the best way is to keep as is then.  But in saas-16.3 we can just remove all migration scripts then.  (so only needed here)

File: addons/l10n_ec/data/account_tax_template_withhold_profit_data.xml [hunk 5]
```diff
@@ -1509,6 +1520,538 @@
...
+                    'minus_report_expression_ids': [ref('tax_report_line_103_303_tag')],
+                }),
+                (0,0, {
+                    'repartition_type': 'tax',
+                    'account_id': ref('l10n_ec.ret_ir_3x100'),
+                    'minus_report_expression_ids': [ref('tax_report_line_103_353_tag')],
+                })]"/>
+        </record>
+
+<!--        <record id="tax_withhold_profit_3482" model="account.tax.template">--><!--  TODO tax commented because it collides with big contributor tax code -->
```
Comment (jco-odoo) [inline addons/l10n_ec/data/account_tax_template_withhold_profit_data.xml:1724]: Better to simply remove.

File: addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def update_account_tax_repartition_line(cr, env):
+    """
+    Update the account in the tax repartition line for the taxes
+    """
+    companies = env['res.company'].search([('chart_template_id', '=', env.ref('l10n_ec.l10n_ec_ifrs').id)])
+    for company in companies:
+        taxes_to_update = [
+            (f'{company.id}_tax_withhold_profit_309', f'{company.id}_ret_ir_1_75x100'),
```
Comment (jco-odoo) [inline addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py:67]: should we create it on the fly if it does not exist?

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**